qiskit/qobj/qasm_qobj.py

```diff
@@ -0,0 +1,482 @@
+                instruction operates on
+            register (list): If a ``measure`` instruction this is a list
+                of `int`s containing the list of register slots in which to
+                store the measurement results (must be the same length as
+                qubits). If a ``bfunc`` instruction this is a single `int`
+                of the register slot in which to store the result.
+            memory (list): If a ``measure`` instruction this is a list
+                of `int`s containing the list of memory slots to store the
+                measurement results in (must be the same length as qubits).
+                If a ``bfunc`` instruction this is a single `int` of the
+                memory slot to store the boolean function result in.
+            condition (tuple): A tuple of the form ``(int, int)`` where the
+                first `int` is the control register and the second `int` is
+                the control value if the gate has a condition.
+            conditional (int):  The register index of the condition
+            label (str): An optional label assigned to the instruction
+            mask (int): For a ``bfunc`` instruction the hex value which is
+                applied as an ``AND`` to the register bits.
+            relation (str): Relational  operator  for  comparing  the  masked
+                register to the `val` kwarg. Can be either ``==`` (equals) or
+                ``!=`` (not equals).
+            val (int): Value to which to compare the masked register. In other
+                words, the output of the function is (``register AND mask)
+                relation val.
+        """
+        super(QasmQobjInstruction, self).__init__()
+        self.name = name
+        if params is not None:
+            self.params = params
+        if qubits is not None:
+            self.qubits = qubits
+        if register is not None:
+            self.register = register
+        if memory is not None:
+            self.memory = memory
+        if condition is not None:
+            self._condition = condition
+        if conditional is not None:
+            self.conditional = conditional
+        if label is not None:
+            self.label = label
+        if mask is not None:
+            self.mask = mask
+        if relation is not None:
+            self.relation = relation
+        if val is not None:
+            self.val = val
+
+    def to_dict(self):
+        """Return a dictionary format representation of the Instruction.
+
+        Returns:
+            dict: The dictionary form of the QasmQobjInstruction.
+        """
+        out_dict = {'name': self.name}
+        for attr in ['params', 'qubits', 'register', 'memory', '_condition',
+                     'conditional', 'label', 'mask', 'relation', 'val']:
+            if hasattr(self, attr):
+                out_dict[attr] = getattr(self, attr)
+
+        return out_dict
```

Comment:
if you make this class a child of `SimpleNamespace` you should be able to do just `return self.__dict__`

Reply:
That's true, but this class doesn't accept arbitrary key value pair fields so I'd prefer to not use simple namespace for this. I'm actually probably going to add `__slots__` here to speed up access a bit which is even more fixed.